### PR TITLE
[uart] Reduce RX FIFO depth from 128B to 64B

### PR DESCRIPTION
**Pull request by msfschaffner**
After more discussion with customers, it became clear that a 64B FIFO size is sufficient.

### CODE REVIEW

**Comment by msfschaffner** [comment]
Can someone run 
```
util/dvsim/dvsim.py -t vcs hw/top_earlgrey/dv/top_earlgrey_sim_cfgs.hjson --select-cfgs uart -i all_once
```
on this PR and post the results? I lost access to my VM for some reason...

If this is all-green except for `uart_stress_all_with_rand_reset` this can be merged (see also https://github.com/lowRISC/opentitan/pull/22026#issuecomment-1996322346).

**Comment by andreaskurth** [comment]
I'm running the regression asked for above; will post the results here.

**Comment by msfschaffner** [comment]
I fixed something and reran the full nightly regression.
There are some sporadic interrupt failures that should be looked at for a V2 signoff, but all tests pass with >90% of seeds except for that stress-all test that is known to fail.
I think we can move forward with this change and patch things up later for the V2 signoff.

File: hw/ip/uart/README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ top level system.
 - 2-pin full duplex external interface
 - 8-bit data word, optional even or odd parity bit per byte
 - 1 stop bit
-- 128 x 8b RX buffer
+- 64 x 8b RX buffer
```
Comment (vogelpi) [inline hw/ip/uart/README.md:23]: Thanks for aligning that!

File: hw/ip/uart/dv/env/uart_scoreboard.sv [hunk 2]
```diff
@@ -248,7 +248,7 @@ class uart_scoreboard extends cip_base_scoreboard #(.CFG_T(uart_env_cfg),
             if (ral.ctrl.slpbk.get_mirrored_value()) begin
               // if sys loopback is on, tx item isn't sent to uart pin but rx fifo
               uart_item tx_item = tx_q.pop_front();
-              if (rx_enabled && (rx_q.size < TxFifoDepth)) begin
+              if (rx_enabled && (rx_q.size < RxFifoDepth)) begin
```
Comment (vogelpi) [inline hw/ip/uart/dv/env/uart_scoreboard.sv:251]: This looks like a drive-by bug fix to me. Correct?

File: sw/device/lib/dif/dif_uart.h [hunk 3]
```diff
@@ -98,11 +98,7 @@ typedef enum dif_uart_watermark {
   /**
    * Indicates a sixty-four-byte watermark.
```
Comment (vogelpi) [inline sw/device/lib/dif/dif_uart.h:99]: Nit: this comment is outdated now.

Comment (vogelpi) [reply sw/device/lib/dif/dif_uart.h:99]: But to be honest, I would leave it as is.

**Review by vogelpi — APPROVED** [review]
LGTM, thanks @msfschaffner !

File: hw/ip/uart/dv/env/uart_scoreboard.sv (hunk 2 above)
Comment (msfschaffner) [reply hw/ip/uart/dv/env/uart_scoreboard.sv:251]: yes, I think I introduced that bug when I separated the rx/tx depth parameters

**Review by moidx — APPROVED** [review]

**Comment by vogelpi** [comment]
I've now rebased this on master as I saw the same unrelated chip-level test failing repeatedly.
